### PR TITLE
feat: Added ability to rollback a helm release

### PR DESCRIPTION
**Pull request by manish-agrawal-ai**
# Description
1) Functionality added to rollback a helm release
2) Helm app Deployment history API response change to support rollback helm release feature

Fixes # https://github.com/devtron-labs/devtron/issues/1281

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
By hitting new Rollback release APIs and helm app deployment history APIs (for both full and hyperion mode)

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: api/openapi/openapiClient/docs/DefaultApi.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
+# \DefaultApi
```
Comment (nishant-d) [inline api/openapi/openapiClient/docs/DefaultApi.md:1]: are we committing autogen doc in the repo?

File: pkg/appStore/deployment/AppStoreDeploymentService.go [hunk 2]
```diff
@@ -615,3 +619,53 @@ func (impl AppStoreDeploymentServiceImpl) createEnvironmentIfNotExists(installAp
...
+	var valuesYaml string
+	var success bool
+	if installedApp.AppOfferingMode == util2.SERVER_MODE_HYPERION {
+		valuesYaml, success, err = impl.appStoreDeploymentHelmService.RollbackRelease(ctx, installedApp, request.GetVersion())
+		if err != nil {
+			impl.logger.Errorw("error while rollback helm release", "error", err)
+			return false, err
+		}
+	} else {
+		// TODO : handle acd
```
Comment (nishant-d) [inline pkg/appStore/deployment/AppStoreDeploymentService.go:650]: ???

File: api/openapi/openapiClient/api/openapi.yaml [hunk 3]
```diff
@@ -0,0 +1,77 @@
+openapi: 3.0.3
```
Comment (nishant-d) [inline api/openapi/openapiClient/api/openapi.yaml:1]: we put specs in `spec` folder

**Review by nishant-d — CHANGES_REQUESTED** [review]

**Review by vikramdevtron — COMMENTED** [review]
unit cases for rollback

File: api/openapi/openapiClient/docs/DefaultApi.md (hunk 1 above)
Comment (manish-agrawal-ai) [reply api/openapi/openapiClient/docs/DefaultApi.md:1]: Will discuss this

File: pkg/appStore/deployment/AppStoreDeploymentService.go (hunk 2 above)
Comment (manish-agrawal-ai) [reply pkg/appStore/deployment/AppStoreDeploymentService.go:650]: Devtron chart's app rollback feature (Just like hyperion helm app) - to be code by Vikram.

File: api/openapi/openapiClient/api/openapi.yaml (hunk 3 above)
Comment (manish-agrawal-ai) [reply api/openapi/openapiClient/api/openapi.yaml:1]: Will discuss this

File: api/openapi/openapiClient/docs/DefaultApi.md (hunk 1 above)
Comment (manish-agrawal-ai) [reply api/openapi/openapiClient/docs/DefaultApi.md]: moved to spec folder.

**Comment by manish-agrawal-ai** [comment]
Unit testcase is not required. API testcases are needed.

**Comment by vikramdevtron** [comment]
@manish-agrawal-ai can we add template for this bug/feature ?

**Review by nishant-d — APPROVED** [review]

**Pull request merged**